lab_2_retrieval_w_bm25/main.py

```diff
@@ -47,6 +75,17 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(documents, list) or len(documents) == 0:
+        return None
+    all_tokens = []
+    for tokens in documents:
+        if not isinstance(tokens, list):
+            return None
+        for elem in tokens:
+            if not isinstance(elem, str):
+                return None
+        all_tokens.extend(tokens)
```

Comment:
try to simplify and to use list comprehensions

Reply:
done